libraries/AP_Relay/AP_Relay.cpp

```diff
@@ -358,6 +397,18 @@ bool AP_Relay::arming_checks(size_t buflen, char *buffer) const
             }
             return false;
         }
+
+        // Each pin can only be used by a single relay
```

Comment:
i think this is preventing a harmless and valid config, for example strobe light on both RC option and arming

Reply:
I don't mind as much letting 2 different functions set it, but the net effect is a very unpredictable system. If you are using anything more complicated (ie starter/ice control) having 2 things fight over the pin is catastrophic and should be blocked. It also means that the relay report as sent out gets more confusing, I'd prefer to keep the functionality as it stands, as it blocks some harmful setups.

Reply:
A middle ground would be to bock useing a pin twice for a "named" function but allow it for "relay" function. So you could not have ICE starter and parachute share a pin, but you could have to relay functions sharing.

I really can't see any genuine existing case where using a pin twice is a good idea. There are lots of cases where it can end up in very bad things. EG parachute and relay from RC option triggering the same thing.

Reply:
I have dropped this pre-arm from this PR, but it will be back in its own PR.